tools/Environments/DevHome.Environments/ViewModels/OperationsViewModel.cs

```diff
@@ -46,12 +51,22 @@ public partial class OperationsViewModel : IEquatable<OperationsViewModel>
 
     private Action? DevHomeAction { get; }
 
-    public OperationsViewModel(string name, string icon, Func<string, Task<ComputeSystemOperationResult>> command, ComputeSystemOperations computeSystemOperation)
+    private WinUIEx.WindowEx? _windowEx;
+
+    private StringResource _stringResource = new("DevHome.Environments.pri", "DevHome.Environments/Resources");
+
+    public OperationsViewModel(
+        string name,
+        string icon,
+        Func<string, Task<ComputeSystemOperationResult>> command,
+        ComputeSystemOperations computeSystemOperation,
+        WinUIEx.WindowEx? windowEx = null)

```

Comment:
It seems like this is the version of the constructor that should be used if there could be a Delete action on it later. Is there some way we could enforce this in the future?

Reply:
The confirmation popup code checks for null so I think that is enough. Would a separate constructor be better?

Reply:
I meant more like, can they use a different constructor, do the delete action, and not see the popup? The null check will prevent a crash, but could the user still find themselves in a situation where they're not warned?

Reply:
I don't think so, since this is where the action is called from. And all future changes will happen here, and should not miss code in the relevant function.